### ISSUE TITLE
`volunteer_summary` exports everything in spreadsheet regardless of definition file

### ISSUE THREAD

**Issue by jlumbroso** [josephlou5/roseingrave#12]
When we created the volunteer sheets, this person was supposed to do certain sonatas. But later it turned out they didn't have time to do the work. So I removed the sonatas from the volunteer definition file. But since the spreadsheets had already been created, when I do `volunteer_summary`, the empty are exported.

In a sense that's good — you export the data in the sheets no matter what. In another sense, maybe it would be good to have an option to either take all that's in the spreadsheet, or only the stuff explicitly requested in the configuration file.

![Screen Shot 2022-11-02 at 9 25 58 PM](https://user-images.githubusercontent.com/11770469/199632005-0e759359-f469-4584-93e3-fc3418ad04a3.png)

![Screen Shot 2022-11-02 at 9 26 09 PM](https://user-images.githubusercontent.com/11770469/199632012-d77e44fc-7b5c-4487-a3b4-5d6da30a0654.png)

**Comment by josephlou5** [comment]
The reason for this is because `volunteer_summary` (the command that actually exports the sheets) does not look at the `volunteer_definitions.json` file. Thus, it does not know which pieces the volunteer has, and just exports all the sheets in each spreadsheet.

Similarly, would you want an option for `export_master` to only export the pieces seen in `piece_definitions.json`?

**Comment by jlumbroso** [comment]
@josephlou5 Just to be clear, I think the way it works now is super useful, it should be a mode, maybe
```
--exportAll    Exports all spreadsheet content, regardless of whether the tab/piece is in the volunteer definition file.
```
The way you were doing it is very sane, because it provides the option to hover up all the data — but I think it's also useful to be able to restrict the scope, for exactly the situation I'm describing.

Or maybe the lesson here is that we could have a:
```
--ignoreEmpty   Ignores any column that does not contain any user entry.
```
Of course these would all make sense in `export_master` too, yes. What do you think?

**Comment by josephlou5** [comment]
@jlumbroso See [the commit](https://github.com/josephlou5/roseingrave/commit/07764688e4dfe89cde554e820d828c372a1da0ca#diff-d5c18e1270eb0b47ca15b51b818baeb95524a15253f8afb3098ff78dec7a0decR137) I just made in the `export-known` branch.

The default behavior is to keep doing what the code currently does. If the `-ek`/`--export-known-only` option is given, only recognized volunteers and pieces will be exported. (The default simply reads all the spreadsheets from `spreadsheets.json` and processes all sheets in every spreadsheet.) I believe this is behavior you are looking for.

With this, we can also add warning messages for missing pieces.

**Comment by jlumbroso** [comment]
@josephlou5 This is exactly the right solution! Thanks for doing triage on my above post, and figuring out what the right delta to implement was. I particularly appreciate how you ensure to take care of edge cases gracefully, and you focus so much on error reporting.

I think the above-par error reporting will be very important in ensuring Roseingrave has a large audience. The error reporting is what will allow this to survive beyond us.

**Comment by josephlou5** [comment]
Resolved with commit 1445dc8eec709b3ecbb5cec8797e11184b8d4824.
